Assets/_BForBoss/_Utility/Scripts/ObjectPooler.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using UnityEngine;
+#if UNITY_2021_1_OR_NEWER
+using UnityEngine.Pool;
+#else
+using UnityEngine.SceneManagement;
+using System.Collections.Generic;
+#endif
+
+namespace Perigon.Utility
+{
+    public class ObjectPooler<T> where T : Component
+    {
+		protected string name;
+
+#if UNITY_2021_1_OR_NEWER
+		protected IObjectPool<T> objectPool;
+#else
+		protected Scene poolScene;
+		protected List<T> pool = null;
+#endif
+
+		protected Func<T> createFunc;
+		protected Action<T> actionOnGet;
+		protected Action<T> actionOnRelease;
+
+        public ObjectPooler(string poolName, Func<T> createFunc, Action<T> actionOnGet, Action<T> actionOnRelease)
+        {
+            this.name = poolName;
+            this.createFunc = createFunc;
+            this.actionOnGet = actionOnGet;
+            this.actionOnRelease = actionOnRelease;
+
+#if UNITY_2021_1_OR_NEWER
+			objectPool = new ObjectPool<T>(createFunc, actionOnGet, actionOnRelease);
+#endif
+		}
+
+        public T Get()
+		{
+
+#if UNITY_2021_1_OR_NEWER
+			return objectPool.Get();
+#else
+			T instance;
+			if (pool == null)
+			{
+				CreatePools();
+			}
+			int lastIndex = pool.Count - 1;
+			if (lastIndex >= 0)
+			{
+				instance = pool[lastIndex];
+				pool.RemoveAt(lastIndex);
+			}
+			else
+			{
+				instance = createFunc();
+				SceneManager.MoveGameObjectToScene(
+					instance.gameObject, poolScene
+				);
+			}
+			actionOnGet?.Invoke(instance);
+			return instance;
+#endif
+		}
+
+		public void Reclaim(T toRecycle)
+		{
+#if UNITY_2021_1_OR_NEWER
+			objectPool.Release(toRecycle);
+#else
+			if (pool == null)
+			{
+				CreatePools();
+			}
+			actionOnRelease?.Invoke(toRecycle);
+			pool.Add(toRecycle);
+#endif
+		}
+
+#if !UNITY_2021_1_OR_NEWER
+		void CreatePools()
+		{
+			pool = new List<T>();
+
+			if (Application.isEditor)
```

Comment:
How come this section is only run in Editor?

Reply:
Its specifically to support domain reload in the editor. If the domain reload triggers, the list gets lost, so this just handles repopulating it. Come to think of it I should wrap this in editor compiler flags

Reply:
Does it still need `if (Application .isEditor)`?

Reply:
Good catch I'll remove that